### ISSUE TITLE
Data generation during the pre-training process

### ISSUE THREAD

**Issue by liuwei13213** [PriorLabs/TabPFN#280]
### Describe the issue linked to the documentation

In train.py, I see that data generation is done in fast_gp.py, but the data generated in fast_gp.py does not seem to support multi-classification because the generated labels are not discrete values. Is multi-classification completed in other Dataloaders? Can the source code be provided?

**Comment by liu-qingyuan** [comment]
How to get the v1 data generation code. It seems that the author will not provide the pre generated code for v2 version data in the short term. [Link](https://github.com/PriorLabs/TabPFN/issues/157)

**Comment by liu-qingyuan** [comment]
Is the data generation method for v1 and v2 consistent, both using a prior architecture that combines SCM and BNN?

**Comment by liu-qingyuan** [comment]
> How to get the v1 data generation code. It seems that the author will not provide the pre generated code for v2 version data in the short term. [Link](https://github.com/PriorLabs/TabPFN/issues/157)

I have found the address for version v1. [HERE](https://github.com/PriorLabs/TabPFN/tree/tabpfn_v1)

**Comment by noahho** [comment]
We don’t currently plan to release full pre-training code for TabPFN v2. However, you can still study and adapt the pre-training setup from v1, which is available on the [tabpfn_v1 branch](https://github.com/PriorLabs/TabPFN/tree/tabpfn_v1/tabpfn). In particular, [this section of train.py](https://github.com/PriorLabs/TabPFN/blob/bef0cf257f3c4c519a2f8e301e0b81a26ae73aff/tabpfn/train.py#L279) shows how the prior is constructed, which is the basis for generating training data.

If your main goal is to adapt TabPFN to your own dataset rather than fully re-pretraining, you may find fine-tuning more practical. We provide an example script here: [finetune_classifier.py](https://github.com/PriorLabs/TabPFN/blob/main/examples/finetune_classifier.py).
